packages/react-native/ReactAndroid/src/test/java/com/facebook/react/modules/timing/TimingModuleTest.kt

```diff
@@ -0,0 +1,265 @@
+import com.facebook.react.modules.core.ReactChoreographer.CallbackType
+import com.facebook.react.modules.core.TimingModule
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.Before
+import org.junit.Ignore
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.ArgumentMatchers.any
+import org.mockito.ArgumentMatchers.eq
+import org.mockito.Mockito.doAnswer
+import org.mockito.Mockito.mock
+import org.mockito.Mockito.reset
+import org.mockito.Mockito.verify
+import org.mockito.Mockito.verifyNoMoreInteractions
+import org.mockito.Mockito.`when` as mockWhen
+import org.mockito.invocation.InvocationOnMock
+import org.mockito.stubbing.Answer
+import org.powermock.api.mockito.PowerMockito
+import org.powermock.core.classloader.annotations.PowerMockIgnore
+import org.powermock.core.classloader.annotations.PrepareForTest
+import org.powermock.modules.junit4.rule.PowerMockRule
+import org.robolectric.RobolectricTestRunner
+
+@[
+PrepareForTest(Arguments::class, SystemClock::class, ReactChoreographer::class)
+PowerMockIgnore("org.mockito.*", "org.robolectric.*", "androidx.*", "android.*")
+RunWith(RobolectricTestRunner::class)]
+class TimingModuleTest {
+  companion object {
+    const val FRAME_TIME_NS = 17 * 1000 * 1000
+  }
+
+  private lateinit var mTimingModule: TimingModule
+  private lateinit var mReactChoreographerMock: ReactChoreographer
+  private lateinit var mPostFrameCallbackHandler: PostFrameCallbackHandler
+  private lateinit var mIdlePostFrameCallbackHandler: PostFrameIdleCallbackHandler
+  private var mCurrentTimeNs = 0L
+  private lateinit var mJSTimersMock: JSTimers
+
+  @get:Rule val powerMockRule = PowerMockRule()
+
+  @Before
+  fun prepareModules() {
+    PowerMockito.mockStatic(Arguments::class.java)
+    mockWhen(Arguments.createArray()).thenAnswer {
+      return@thenAnswer JavaOnlyArray()
+    }
+
+    PowerMockito.mockStatic(SystemClock::class.java)
+    mockWhen(SystemClock.uptimeMillis()).thenAnswer {
+      return@thenAnswer mCurrentTimeNs / 1000000
+    }
+    mockWhen(SystemClock.currentTimeMillis()).thenAnswer {
+      return@thenAnswer mCurrentTimeNs / 1000000
+    }
+    mockWhen(SystemClock.nanoTime()).thenAnswer {
+      return@thenAnswer mCurrentTimeNs
+    }
+
+    mReactChoreographerMock = mock(ReactChoreographer::class.java)
+    PowerMockito.mockStatic(ReactChoreographer::class.java)
+    mockWhen(ReactChoreographer.getInstance()).thenAnswer { mReactChoreographerMock }
+
+    val reactInstance = mock(CatalystInstance::class.java)
+    val reactContext = mock(ReactApplicationContext::class.java)
+    mockWhen(reactContext.catalystInstance).thenReturn(reactInstance)
+    mockWhen(reactContext.hasActiveReactInstance()).thenReturn(true)
+
+    mPostFrameCallbackHandler = PostFrameCallbackHandler()
+    mIdlePostFrameCallbackHandler = PostFrameIdleCallbackHandler()
+
+    doAnswer(mPostFrameCallbackHandler)
+      .`when`(mReactChoreographerMock)
```

Comment:
can you static import this `when` as well?

Reply:
It is not an import. It's an interface function (see `Stubber` type which is returned by `doAnswer) so we are stuck with the naming. 

But I think we can work around that by starting with `whenever`... will test it out.